src/main/java/com/minecolonies/coremod/entity/ai/citizen/smelter/EntityAIWorkSmelter.java

```diff
@@ -150,11 +163,19 @@ private AIState smeltStuff()
             final ItemStack stack = new InvWrapper(worker.getInventoryCitizen()).extractItem(slot, 1, false);
             final Tuple<ItemStack, Integer> materialTuple = getMaterialAndAmount(stack);
             final ItemStack material = materialTuple.getFirst();
-            if(!ItemStackUtils.isEmpty(material))
+            if (!ItemStackUtils.isEmpty(material))
             {
                 material.setCount(materialTuple.getSecond());
                 material.setItemDamage(0);
                 new InvWrapper(worker.getInventoryCitizen()).setStackInSlot(slot, material);
+                if (stack.isItemEnchanted())
+                {
+                    if (ENCHANTED_BOOK_CHANCE[getOwnBuilding().getBuildingLevel() - 1] < new Random().nextInt(100))
+                    {
+                        final ItemStack book = extractEnchantFromItem(stack);
+                        new InvWrapper(worker.getInventoryCitizen()).insertItem(InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(worker.getInventoryCitizen()),ItemStack::isEmpty),book,false);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 203 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

Reply:
make a line break here on new InvWrapper.